src/frontend/src/icp/components/transactions/IcTransaction.svelte

```diff
@@ -1,74 +1,51 @@
-	import IconConvertFrom from '$lib/components/icons/IconConvertFrom.svelte';
-	import IconConvertTo from '$lib/components/icons/IconConvertTo.svelte';
-	import IconReceive from '$lib/components/icons/IconReceive.svelte';
-	import IconSend from '$lib/components/icons/IconSend.svelte';
-	import TransactionStatusComponent from '$lib/components/transactions/TransactionStatus.svelte';
-	import Amount from '$lib/components/ui/Amount.svelte';
-	import Card from '$lib/components/ui/Card.svelte';
-	import RoundedIcon from '$lib/components/ui/RoundedIcon.svelte';
+	import Transaction from '$lib/components/transactions/Transaction.svelte';
+	import { NANO_SECONDS_IN_SECOND } from '$lib/constants/app.constants';
 	import { modalStore } from '$lib/stores/modal.store';
 	import type { TransactionStatus } from '$lib/types/transaction';
-	import { formatNanosecondsToDate } from '$lib/utils/format.utils';
 
 	export let transaction: IcTransactionUi;
 
 	let type: IcTransactionType;
 	let transactionTypeLabel: string | undefined;
 	let value: bigint | undefined;
-	let timestamp: bigint | undefined;
+	let timestampNanoseconds: bigint | undefined;
 	let incoming: boolean | undefined;
 
-	$: ({ type, typeLabel: transactionTypeLabel, value, timestamp, incoming } = transaction);
+	$: ({
+		type,
+		typeLabel: transactionTypeLabel,
+		value,
+		timestamp: timestampNanoseconds,
+		incoming
+	} = transaction);
 
 	let pending = false;
 	$: pending = transaction?.status === 'pending';
 
 	let status: TransactionStatus;
 	$: status = pending ? 'pending' : 'confirmed';
 
-	let icon: ComponentType;
-	$: icon =
-		['burn', 'approve', 'mint'].includes(type) && pending
-			? IconConvert
-			: ['burn', 'approve'].includes(type)
-				? IconConvertTo
-				: type === 'mint'
-					? IconConvertFrom
-					: type === 'send'
-						? IconSend
-						: IconReceive;
-
 	let amount: bigint | undefined;
 	$: amount = !incoming && nonNullish(value) ? value * -1n : value;
-</script>
-
-<button on:click={() => modalStore.openIcTransaction(transaction)} class="block w-full border-0">
-	<Card>
-		<span class="inline-block first-letter:capitalize"
-			><IcTransactionLabel label={transactionTypeLabel} fallback={type} /></span
-		>
 
-		<RoundedIcon slot="icon" {icon} />
-
-		<svelte:fragment slot="amount">
-			{#if nonNullish(amount)}
-				<Amount amount={BigNumber.from(amount)} />
-			{/if}
-		</svelte:fragment>
-
-		<svelte:fragment slot="description">
-			{#if nonNullish(timestamp)}
-				{formatNanosecondsToDate(timestamp)}
-			{/if}
+	let timestamp: number | undefined;
+	$: timestamp = nonNullish(timestampNanoseconds)
+		? Number(timestampNanoseconds / NANO_SECONDS_IN_SECOND)
+		: undefined;
+</script>
 
-			<TransactionStatusComponent {status} />
-		</svelte:fragment>
-	</Card>
-</button>
+<Transaction
+	on:click={() => modalStore.openIcTransaction(transaction)}
+	styleClass="block w-full border-0"
+	amount={BigNumber.from(amount)}
```

Comment:
for BTC it can still be, according to current implementation. I am not really sure the use case though... maybe @DenysKarmazynDFINITY can give us some idea?

Reply:
I remember now! Thank you for the explanation!

@peterpeterparker does that answer the question and/or solves the doubt?

Reply:
Alright but, I would not modify the behavior in this PR which has for purpose the use of a component not the way we handle the amount.

Reply:
indeed, nothing is changing, the old code was using the component Amount directly:

`<Amount amount={BigNumber.from(amount)} />`

while now I am passing it as prop:

`amount={BigNumber.from(amount)}`

but the definition and the logic of `amount` did not change